routing/control_tower.go

```diff
@@ -305,4 +344,21 @@ func (p *controlTower) notifySubscribers(paymentHash lntypes.Hash,
 		case <-subscriber.quit:
 		}
 	}
+
+	// Notify all subscribers that subscribed to all payments.
+	for index, subscriber := range p.subscribersAllPayments {
+		select {
+		case subscriber.queue.ChanIn() <- event:
+
+		// If subscriber disappeared, remove it from the subscribers
+		// list.
+		case <-subscriber.quit:
+			p.subscribersMtx.Lock()
+			p.subscribersAllPayments = append(
```

Comment:
I am a bit worried about this. We are modifying the array size here but `range` is evaluated only once (at the beginning of the loop, so if you change it's size it will not be reflected in the next iteration).

An [example with playground](https://go.dev/play/p/YpsrnMS8ARX) to see the real behavior + the definition from the [go spec](https://go.dev/ref/spec#For_range) in case I am missing something.

Reply:
Fixed this by using a map and removing based on the key. Should be safe according to [stackoverflow](https://stackoverflow.com/a/23230406/3883866)